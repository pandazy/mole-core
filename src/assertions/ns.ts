import { ns as pNs } from '../ns';

export function ns(context: string): string {
	return pNs(`assertions:${context}`);
}
