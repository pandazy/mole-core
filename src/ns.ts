export function ns(context: string): string {
  return `Mole:${context}`;
}
