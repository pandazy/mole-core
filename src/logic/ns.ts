import { ns as pNs } from '../ns';

export function ns(context: string): string {
	return pNs(`logic:${context}`);
}
