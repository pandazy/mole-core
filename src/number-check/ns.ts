import { ns as pNs } from '../ns';

export function ns(context: string): string {
  return pNs(`numericCheck:${context}`);
}
