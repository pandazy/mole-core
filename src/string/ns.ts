import { ns as pNs } from '../ns';

export function ns(context: string): string {
  return pNs(`string:${context}`);
}
