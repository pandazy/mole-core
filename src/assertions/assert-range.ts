import { assertValue } from './assert-value';
import { isWithin } from '../number-check/is-within';

export const assertRange = (from: number) => (to: number) => (
  context: string,
) => (value: number) =>
  assertValue<number>({
    isValid: isWithin(from)(to),
    failureCause: `out of bounds of ${from}..${to}`,
  })(context)(value);
