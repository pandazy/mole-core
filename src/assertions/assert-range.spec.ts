import { assertRange } from './assert-range';
import { ns } from './ns';

describe(ns('assertRange'), () => {
  it('should assert a specific range', () => {
    expect.assertions(2);

    const assert1to7 = assertRange(1)(7)('test@arg');

    expect(() => assert1to7(0)).toThrowErrorMatchingSnapshot();
    assert1to7(1);
    expect(() => assert1to7(8)).toThrowErrorMatchingSnapshot();
  });
});
