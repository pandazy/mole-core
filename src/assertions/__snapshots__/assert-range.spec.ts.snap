// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mole:assertions:assertRange should assert a specific range 1`] = `
"
[context]: test@arg
[failure cause]: out of bounds of 1..7
[actual value]: 0
[actual value type]: Number"
`;

exports[`Mole:assertions:assertRange should assert a specific range 2`] = `
"
[context]: test@arg
[failure cause]: out of bounds of 1..7
[actual value]: 8
[actual value type]: Number"
`;
